src/gluonts/dataset/parallelized_loader.py

```diff
@@ -196,7 +170,7 @@ def batchify(
     dtype: DType,
     multi_processing: bool,
     single_process_ctx: Optional[mx.Context] = None,
-    variable_length: bool = False,
+    variable_length: bool = True,
```

Comment:
Is the change of default intentional? Why is that?

Reply:
I changed the left padding of the ragged tensors in `MQ-CNN` to be done by the data loader, which I though made more sense.  It looks `variable_length` is only used on `line 128`: 
```
if variable_length and not _is_stackable(data):
         data = pad_arrays(data, axis=0)
```

So updating the default to True, will only result in calling `_is_stackable()`, which doesn't seem computationally expensive because it just looks at the shapes and then for all algorithms would left pad ragged tensors if they have mismatching shapes from time series of different lengths and if they do all have the shape shape `not _is_stackable(data)` will return as `False` and and then `pad_arrays()` won't be called. Is there a way I can pass it as `True` to `batchify` for `MQ-CNN` without updating the default?

 I also wanted to check with you on the prior `pad_arrays` implementation.  I combined it with the one from MQ-CNN and moved to `util.py`, so other models could use the padding, but it seems like in the data loader before it was right padding with zeros instead of left padding and I updated it to left padding. I wasn't sure on why right padding was the default?

Reply:
> I changed the left padding of the ragged tensors in MQ-CNN to be done by the data loader, which I though made more sense. 

As opposed to the instance splitter, you mean?

I think the padding helper function (as well as the `variable_length` option) was out there for the specific use case of temporal point processes, so this change may significantly affect those. I would ask @canerturkmen or @shchur to chip in here.

I think we should avoid relying on the data loader, as much as possible, for model-specific behavior. TPPs constitute a necessary exception here, I think, because of the intrinsic nature of the data they deal with; but in the case of MQCNN the time length is pre-determined, right?

Reply:
Yes, I removed the left padding from the instance splitter.  Yes, makes sense that the ragged tensors were for TPP.  We should confirm with @canerturkmen that the desired behavior for TPP was really right padding instead of left padding with zeros?  I think still keep the padding function in the `util` file so that it can be used in both the loader and MQ-CNN instance splitter, rather than duplicate code.

Yes, I can move the left padding back to the MQ-CNN instance splitter.  It's simple for the `past` target and dynamic features, we simply past to `enc_len = context_length` in the encoder.  I was experimenting with the forking_decoder, where we compute the sliding windows with `as_strided`.  It seems that `as_strided` is efficient but accessing the 3D array `future_feature_dynamic[skip:]` can be slow, so I was testing if padding may be faster, but I think updating the zero array in place is faster.  I will move this back to the instance splitter.  Thanks!